src/RemoteInboxNotifications/StoredStateSetupForProducts.php

```diff
@@ -93,10 +95,14 @@ public static function run_on_transition_post_status( $new_status, $old_status,
 			return;
 		}
 
+		update_option( self::NEW_PRODUCT_COUNT_OPTION_NAME, 1 );
+
 		$stored_state                         = RemoteInboxNotificationsEngine::get_stored_state();
 		$stored_state->there_are_now_products = true;
 		RemoteInboxNotificationsEngine::update_stored_state( $stored_state );
 
 		RemoteInboxNotificationsEngine::run();
+
+		update_option( self::NEW_PRODUCT_COUNT_OPTION_NAME, 0 );
```

Comment:
Not a show stopper, and this could be my lack of understanding of the inbox notification code. But would there also be a way to pass the product count through the `run` or `stored_state` functions instead of using wp option to temporarily store it.
It's not huge, but it always reminds me that another plugin or something could change that in-between.

Reply:
Passing it via `stored_state` is a great idea, I'll do that.

Reply:
Done in 588450c

Reply:
Awesome, looks good, thanks :)